tools/ubuntu-setup.sh

```diff
@@ -0,0 +1,272 @@
+    config_with_sed "CONFIG_KASAN"
+    config_with_sed "CONFIG_KASAN_INLINE"
+
+    # For detection of enabled syscalls and kernel bitness:
+    config_with_sed "CONFIG_KALLSYMS"
+    config_with_sed "CONFIG_KALLSYMS_ALL"
+
+    # Sandbox
+    config_with_sed "CONFIG_NAMESPACES"
+    config_with_sed "CONFIG_UTS_NS"
+    config_with_sed "CONFIG_IPC_NS"
+    config_with_sed "CONFIG_PID_NS"
+    config_with_sed "CONFIG_NET_NS"
+    config_with_sed "CONFIG_CGROUP_PIDS"
+    config_with_sed "CONFIG_MEMCG"
+    config_with_sed "CONFIG_USER_NS"
+
+    # For testing with fault injection enable the following configs
+    # (syzkaller will pick it up automatically):
+    config_with_sed "CONFIG_FAULT_INJECTION"
+    config_with_sed "CONFIG_FAULT_INJECTION_DEBUG_FS"
+    config_with_sed "CONFIG_FAILSLAB"
+    config_with_sed "CONFIG_FAIL_PAGE_ALLOC"
+    config_with_sed "CONFIG_FAIL_MAKE_REQUEST"
+    config_with_sed "CONFIG_FAIL_IO_TIMEOUT"
+    config_with_sed "CONFIG_FAIL_FUTEX"
+
+    config_with_sed "CONFIG_LOCKDEP"
+    config_with_sed "CONFIG_PROVE_LOCKING"
+    config_with_sed "CONFIG_DEBUG_ATOMIC_SLEEP"
+    config_with_sed "CONFIG_PROVE_RCU"
+    config_with_sed "CONFIG_DEBUG_VM"
+    config_with_sed "CONFIG_REFCOUNT_FULL"
+    config_with_sed "CONFIG_FORTIFY_SOURCE"
+    config_with_sed "CONFIG_HARDENED_USERCOPY"
+    config_with_sed "CONFIG_LOCKUP_DETECTOR"
+    config_with_sed "CONFIG_SOFTLOCKUP_DETECTOR"
+    config_with_sed "CONFIG_HARDLOCKUP_DETECTOR"
+    config_with_sed "CONFIG_BOOTPARAM_HARDLOCKUP_PANIC"
+    config_with_sed "CONFIG_DETECT_HUNG_TASK"
+    config_with_sed "CONFIG_WQ_WATCHDOG"
+
+    make O="$KBLD_DIR"
+}
+
+function create_image {
+    cd "$INIT_DIR"
+
+    set -eux
+
+    # Create a minimal Debian distribution in a directory.
+    sudo rm -rf $IMG_BLD_DIR
+    mkdir -p $IMG_BLD_DIR
+    sudo debootstrap --include=openssh-server,curl,tar,gcc,libc6-dev,time,strace,sudo,less,psmisc,selinux-utils,policycoreutils,checkpolicy,selinux-policy-default $IMG_RELEASE $IMG_BLD_DIR
+
+    # Set some defaults and enable promtless ssh to the machine for root.
+    sudo sed -i '/^root/ { s/:x:/::/ }' $IMG_BLD_DIR/etc/passwd
+    echo 'T0:23:respawn:/sbin/getty -L ttyS0 115200 vt100' | sudo tee -a $IMG_BLD_DIR/etc/inittab
+    printf '\nauto eth0\niface eth0 inet dhcp\n' | sudo tee -a $IMG_BLD_DIR/etc/network/interfaces
+    echo '/dev/root / ext4 defaults 0 0' | sudo tee -a $IMG_BLD_DIR/etc/fstab
+    echo 'debugfs /sys/kernel/debug debugfs defaults 0 0' | sudo tee -a $IMG_BLD_DIR/etc/fstab
+    echo 'securityfs /sys/kernel/security securityfs defaults 0 0' | sudo tee -a $IMG_BLD_DIR/etc/fstab
+    echo 'configfs /sys/kernel/config/ configfs defaults 0 0' | sudo tee -a $IMG_BLD_DIR/etc/fstab
+    echo 'binfmt_misc /proc/sys/fs/binfmt_misc binfmt_misc defaults 0 0' | sudo tee -a $IMG_BLD_DIR/etc/fstab
+    echo "kernel.printk = 7 4 1 3" | sudo tee -a $IMG_BLD_DIR/etc/sysctl.conf
```

Comment:
Let's use tools/create-image.sh
It's lots of duplication and it will get out of sync. E.g. I will fix up create-image.sh but can forget to fix up and test this script.